x-pack/plugins/observability_solution/metrics_data_access/common/inventory_models/types.ts

```diff
@@ -235,29 +235,34 @@ export type TSVBMetricModelCreator = (
   interval: string
 ) => TSVBMetricModel;
 
-export const ESBasicMetricAggRT = rt.record(
-  rt.string,
-  rt.union([
-    rt.undefined,
-    rt.type({
-      field: rt.string,
-    }),
-  ])
-);
-
-export const ESDerivativeAggRT = rt.type({
-  derivative: rt.type({
-    buckets_path: rt.string,
-    gap_policy: rt.keyof({ skip: null, insert_zeros: null }),
-    unit: rt.string,
-  }),
-});
+export const isBasicMetricAgg = (
+  agg: unknown
+): agg is Record<string, undefined | Pick<estypes.AggregationsMetricAggregationBase, 'field'>> => {
+  if (agg && typeof agg === 'object' && Object.keys(agg).length !== 0) {
+    for (const key in agg) {
+      if (Object.hasOwn(agg, key)) {
+        return !!(
+          agg[key as keyof typeof agg] === undefined ||
+          (agg[key as keyof typeof agg] as estypes.AggregationsMetricAggregationBase).field
+        );
+      }
+    }
+  }
```

Comment:
>[!TIP]
> Maybe this can be simplified?

```suggestion
 if (typeof agg !== 'object' || agg === null) {
    return false;
  }

  return Object.values(agg).some(
    value => value === undefined || 'field' in (value as estypes.AggregationsMetricAggregationBase)
  );
```